### PR TITLE
AMLS-3785: Check your answers page

### PR DESCRIPTION
**Pull request by elkdanger**

### CODE REVIEW

File: app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala [hunk 1]
```diff
@@ -16,23 +16,53 @@
...
+          model <- updateSubscription
+          route <- OptionT.liftF(router.getRoute(RemoveBusinessTypesSummaryPageId, model))
+        } yield route
+      } getOrElse InternalServerError("Unable to remove the business type")
   }
+
+  private def updateSubscription()
+                                (implicit ac: AuthContext, hc: HeaderCarrier): OptionT[Future, RemoveBusinessTypeFlowModel] = for {
+    model <- OptionT(dataCacheConnector.fetch[RemoveBusinessTypeFlowModel](RemoveBusinessTypeFlowModel.key))
+    _ <- helper.removeFitAndProper(model)
```
Comment (nonimay) [inline app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala]: Where does the different types of section data get removed?

File: app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala [hunk 2]
```diff
@@ -16,23 +16,53 @@
...
+    _ <- helper.removeBusinessMatchingBusinessTypes(model)
+    _ <- helper.removeTradingPremisesBusinessTypes(model)
+  } yield model
```
Comment (nonimay) [inline app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala]: do you clear the flow model from the data cache?

File: app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala (hunk 1 above)
Comment (elkdanger) [reply app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala]: AMLS-3814 has been raised to deal with this issue.

File: app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala (hunk 2 above)
Comment (elkdanger) [reply app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala]: AMLS-3814 has been raised to deal with this issue.

**Pull request merged**